File: Jack.Model/Model/TipoItem.vb
<Serializable()> _
Public Class TipoItem

#Region "Construtor"

    Public Sub New()
        Codigo = 0
        Descricao = String.Empty
    End Sub

    Public Sub New(strDescricao As String)
        Codigo = 0
        Descricao = strDescricao
    End Sub

    Public Sub New(intCodigo As Integer, strDescricao As String)
        Codigo = intCodigo
        Descricao = strDescricao
    End Sub

#End Region

    Public Overridable Property Codigo As Integer
    Public Overridable Property Descricao As String
    Public Overridable Property IsOpcional As String

    Public Overridable ReadOnly Property Opcional As String
        Get
            If IsOpcional = "S" Then
                Return "Opcional"
            Else
                Return "Não"
            End If
        End Get
    End Property

End Class

File: Jack.Model/Model/TipoItem.cs
using Jack.Library.Extensions;
using System;

namespace Jack.Model
{
    [Serializable()]
    public class TipoItem:BaseModel<TipoItem>
    {

        #region "Construtor"

        public TipoItem() : base()
        {
            Codigo = 0;
            descricao = string.Empty;
            isOpcional= string.Empty;
            opcional = string.Empty;
        }

        public TipoItem(string strDescricao)
        {
            Codigo = 0;
            Descricao = strDescricao;
            isOpcional = string.Empty;
            opcional = string.Empty;
        }

        public TipoItem(int intCodigo, string strDescricao)
        {
            Codigo = intCodigo;
            Descricao = strDescricao;
            isOpcional = string.Empty;
            opcional = string.Empty;
        }

        public TipoItem(int intCodigo, string strDescricao, string strIsOpcional)
        {
            Codigo = intCodigo;
            Descricao = strDescricao;
            isOpcional = strIsOpcional;
            opcional = strIsOpcional.ToSimNao();
        }

        #endregion

        private string isOpcional;
        private string opcional;
        private string descricao;

        public virtual string Descricao
        {
            get
            {
                return descricao;
            }
            set
            {
                descricao = value;
            }
        }

        public virtual string IsOpcional
        {
            get
            {
                return isOpcional;
            }
            set
            {
                opcional = IsOpcional.ToSimNao();
                isOpcional = value;
            }
        }

        public virtual string Opcional
        {
            get
            {
                return opcional;
            }
        }

    }
}
